app/src/main/java/com/example/notebook/view/fragments/my_notes/MyNotesFragment.kt

```diff
@@ -8,31 +8,33 @@ import androidx.navigation.findNavController
 import androidx.navigation.fragment.findNavController
 import androidx.recyclerview.widget.LinearLayoutManager
 import com.example.notebook.R
-import com.example.notebook.data.Repositories
-import com.example.notebook.data.note.entities.Note
 import com.example.notebook.databinding.FragmentListNotesBinding
+import com.example.notebook.domain.models.Note
 import com.example.notebook.view.fragments.BaseFragment
 import com.example.notebook.view.fragments.my_notes.adapter.NotesAdapter
 import com.example.notebook.view.interfaces.NotesAdapterListener
+import dagger.hilt.android.AndroidEntryPoint
 
+@AndroidEntryPoint
 class MyNotesFragment: BaseFragment<FragmentListNotesBinding>(FragmentListNotesBinding::inflate),
 SearchView.OnQueryTextListener {
 
     private val viewModel: MyNoteViewModel by viewModels()
 
-    private val notesAdapter = NotesAdapter(
-        listener = object : NotesAdapterListener {
-            override fun onClick(note: Note) {
-                val direction =
-                    MyNotesFragmentDirections.actionMyNotesFragmentToDetailsNoteFragment(note)
-                findNavController().navigate(direction)
+    private val notesAdapter by lazy {
+        NotesAdapter(
+            listener = object : NotesAdapterListener {
```

Comment:
I suggest you to be more specific with the naming of the listener since you can have several of those